docs/content/docs/data-structures/task-manager.md

```diff
@@ -1,5 +1,135 @@
+
+
+- `volunteerForTask(taskId)` -- Adds the client to the task queue **once**. It returns a promise that resolves `true` if the client is assigned the task and `false` if the task was completed by another client. It will throw an error if the client disconnects while in queue.
+- `subscribeToTask(taskId)` -- Will continuously add the client to the task queue. Does not return a value, and will therefore require lisnening to [events](#events) to determine if the task is assigned, lost, or completed.
+- `subscribed(taskId)` -- Returns a boolean to indicate if the client is subscribed to the task.
+- `complete(taskId)` -- Will release all clients from the task queue, including the currently assigned client.
+- `abandon(taskId)` -- Exits the queue and releasing the task if currently assigned. Will also unsubscribe from the task (if subscribed).
+- `queued(taskId)` -- Returns a boolean to indicate if the client is in the task queue (being assigned a task is still considered queued).
+- `assigned(taskId)` -- Returns a boolean to indicate if the client is assigned the task.
+
+### Events
+
+`TaskManager` is an `EventEmitter`, and will emit events when a task is assigned to the client or released. Each of the below events fire with an event listener that contains a callback argument `taskId`. This represents the task for which the event was fired for.
+
+- `assigned` -- Fires when the client reaches the top of the task queue and is assigned the task.
+- `lost` -- Fires when the client disconnects while being assigned a task and is removed from the queue.
+- `completed` -- Fires when the assigned client successfully calls `complete()`.
+
+### Creation
+
+To create a `TaskManager`, call the static create method:
+
+```typescript
+const taskManager = TaskManager.create(this.runtime, id);
+```
+
+## Examples
+
+### Basic Example -- `volunteerForTask()`
+
+The following is a basic example for `volunteerForTask()`. Note that we check the `boolean` return value from the promise to ensure that the task was not completed by another client.
+
+```typescript
+const myTaskId = "myTaskId";
+
+taskManager.volunteerForTask(myTaskId)
+  .then((isAssigned: boolean) => {
+    if (isAssigned) {
+      console.log("Assigned task.");
+
+      // We setup a listener in case we lose the task assignment while executing the code.
+      const onLost = (taskId: string) => {
+        if (taskId === myTaskId) {
+          // The task assignment has been lost, therefore we should halt execution.
+          stopExecutingTask();
+        }
+      };
+      taskManager.on("lost", onLost);
+
+      // Now that we are assigned the task we can begin executing the code.
+      executeTask()
+        .then(() => {
+          // We should remember to turn off the listener once we are done with it.
+          taskManager.off("lost", onLost);
```

Comment:
Shouldn't `complete` be called here too? If not, how is "complete" triggered?

Reply:
I think I wanted to imply that `complete()` is being called at the end of `executeTask()`, but that's definitely not clear and can be done inside the `then` block as well. I'll add a note here and inside the `subscribeToTask()` example as well.